### PR TITLE
Feature/compound borrow fee calculator

### PR DESCRIPTION
**Pull request by MussinaLin**

### CODE REVIEW

File: test/fees/CompoundBorrowFeeCalculator.t.sol [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        vm.label(address(borrowFeeCalculator), 'BorrowFeeCalculator');
+        vm.label(C_USDCV3, 'cUSDCv3');
+    }
+
+    function testChargeBorrowFee(uint256 amount, uint256 feeRate) external {
+        feeRate = bound(feeRate, 0, BPS_BASE - 1);
+
+        // Calculate borrow limit, assume ETH price is 1000
+        IComet.AssetInfo memory assetInfo = IComet(comet).getAssetInfoByAddress(collateral);
+        uint256 collateralRatio = assetInfo.borrowCollateralFactor;
```
Comment (JavanC) [inline test/fees/CompoundBorrowFeeCalculator.t.sol:103]: The collateral ratiao is 1e18 format?

File: test/fees/CompoundBorrowFeeCalculator.t.sol [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+contract CompoundBorrowFeeCalculatorTest is Test {
+    event FeeCharged(address indexed token, uint256 amount, bytes32 metadata);
+
+    address public constant NATIVE = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    address public constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+    address public constant C_USDCV3 = 0xc3d688B66703497DAA19211EEdff47f25384cdc3;
+    address public constant DUMMY_TO_ADDRESS = address(0);
+    bytes32 public constant META_DATA = bytes32(bytes('compound-v3:borrow'));
+    bytes4 public constant COMPOUND_BORROW_SELECTOR =
+        bytes4(keccak256(bytes('withdrawFrom(address,address,address,uint256)')));
```
Comment (JavanC) [inline test/fees/CompoundBorrowFeeCalculator.t.sol:44]: Curious how to distinguish between the user wants Withdraw and Borrow?

File: test/fees/CompoundBorrowFeeCalculator.t.sol (hunk 1 above)
Comment (MussinaLin) [reply test/fees/CompoundBorrowFeeCalculator.t.sol:103]: yes, reference https://github.com/compound-developers/compound-3-developer-faq/blob/master/examples/get-cf-examples.js#L70

File: test/fees/CompoundBorrowFeeCalculator.t.sol [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    address public borrowFeeCalculator;
+
+    // Empty arrays
+    address[] public tokensReturnEmpty;
+    IParam.Input[] public inputsEmpty;
+
+    address public comet = C_USDCV3;
+    address public collateral = WETH;
+    uint256 public collateralAmount = 3000 ether;
+    address public baseToken = IComet(comet).baseToken();
```
[the remark this answers is not in the document]
Comment (MussinaLin) [reply test/fees/CompoundBorrowFeeCalculator.t.sol:61]: Will move to Ln.47

File: test/fees/CompoundBorrowFeeCalculator.t.sol (hunk 2 above)
Comment (MussinaLin) [reply test/fees/CompoundBorrowFeeCalculator.t.sol:44]: It's hard to distinguish user want to Withdraw or Borrow. The discussion conclusion is [here](https://app.asana.com/0/0/1204244120891484/1204519489763734/f)

File: script/DeployAll.s.sol [hunk 4]
```diff
@@ -24,7 +25,8 @@ contract DeployAll is
     DeployMakerDrawFeeCalculator,
     DeployNativeFeeCalculator,
     DeployPermit2FeeCalculator,
-    DeployTransferFromFeeCalculator
+    DeployTransferFromFeeCalculator,
+    DeployCompoundBorrowFeeCalculator
```
[the remark this answers is not in the document]
Comment (MussinaLin) [reply script/DeployAll.s.sol:29]: Sorry didn't notice that, will fix

File: script/DeployCompoundBorrowFeeCalculator.s.sol
[the remark this answers is not in the document]
Comment (MussinaLin) [reply script/DeployCompoundBorrowFeeCalculator.s.sol:1]: OK

File: test/fees/CompoundBorrowFeeCalculator.t.sol (hunk 1 above)
Comment (JavanC) [reply test/fees/CompoundBorrowFeeCalculator.t.sol:103]: I see. Originally I thought why divide by 1e18

File: test/fees/CompoundBorrowFeeCalculator.t.sol (hunk 2 above)
Comment (JavanC) [reply test/fees/CompoundBorrowFeeCalculator.t.sol:44]: I see.

File: script/DeployCompoundBorrowFeeCalculator.s.sol
[the remark this answers is not in the document]
Comment (MussinaLin) [reply script/DeployCompoundBorrowFeeCalculator.s.sol:1]: A specific [commit](https://github.com/dinngo/composable-router-contract/pull/64/commits/130a031b252173b6413f2045ba52c9ae13485533) for rename file.

File: src/fees/CompoundBorrowFeeCalculator.sol [hunk 5]
```diff
@@ -0,0 +1,31 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.0;
+
+import {FeeCalculatorBase} from './FeeCalculatorBase.sol';
+import {IFeeCalculator} from '../interfaces/IFeeCalculator.sol';
+import {IParam} from '../interfaces/IParam.sol';
+
+contract CompoundBorrowFeeCalculator is IFeeCalculator, FeeCalculatorBase {
```
[the remark this answers is not in the document]
Comment (MussinaLin) [reply src/fees/CompoundBorrowFeeCalculator.sol]: Added

**Comment by JavanC** [comment]
LGTM

**Comment by zodahu** [comment]
LGTM

**Comment by zodahu** [comment]
LGTM

**Pull request merged**